### PR TITLE
CMake: Add CMake support

### PR DESCRIPTION
**Pull request by rajkan01**
- Added `CMakeLists.txt file` to Mbed client for Azure cloud so that it can able build via CMake.

@0xc0170 @hugueskamba @LDong-Arm @evedon

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        azure-iot-sdk-c/iothub_client/src/iothub_transport_ll_private.c
+        azure-iot-sdk-c/iothub_client/src/iothubtransport.c
+        azure-iot-sdk-c/iothub_client/src/message_queue.c
+        azure-iot-sdk-c/iothub_client/src/iothubtransportmqtt.c
+        azure-iot-sdk-c/iothub_client/src/version.c
+        azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c
+        azure-iot-sdk-c/iothub_client/src/iothub_module_client_ll.c
+        azure-iot-sdk-c/iothub_client/src/iothub_message.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_edge.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_core.c
```
Comment (LDong-Arm) [inline CMakeLists.txt:111]: Please also add the provisioning client (added in #11) to sources and includes, thanks

**Review by LDong-Arm — CHANGES_REQUESTED** [review]
@rajkan01 Thanks, looks a lot better than the old build system's `.mbedignore` approach.
We'd wait until we have found a solution for CMake's macro string-quoting issue, so we can test this with mbed-os-example-for-azure.

File: CMakeLists.txt (hunk 1 above)
Comment (rajkan01) [reply CMakeLists.txt:115]: @LDong-Arm I have added provisioning client source into CMake, please re-review

**Comment by rajkan01** [comment]
> 
> 
> @rajkan01 Thanks, looks a lot better than the old build system's `.mbedignore` approach.
> We'd wait until we have found a solution for CMake's macro string-quoting issue, so we can test this with mbed-os-example-for-azure.
@LDong-Arm Once you are fine with the changes, we can proceed to merge this PR as I have already tested these changes

**Review by LDong-Arm — APPROVED** [review]

**Pull request merged**